src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -111,4 +128,109 @@ private void addListingToBusinesses(PlacesSearchResult place) {
     allBusinesses.add(new Listing(name, formattedAddress, 
           placeLocation, rating, photos, types));
   }
-}
\ No newline at end of file
+
+  public void checkNumberOfLocationsOfBusiness() {
+    GeoApiContext context = new GeoApiContext.Builder()
+            .apiKey(KEY)
+            .build();
+    businesses =  allBusinesses.iterator(); 
+    try {
+      while (businesses.hasNext()) {
+        currentBusiness = businesses.next();
+        TextSearchRequest request = new TextSearchRequest(context)
+                                            .query(currentBusiness.getName())
+                                            .location(latLng)
+                                            .radius(50000);
+        PlacesSearchResult[] similarBusinessesInTheArea = request
+                                                            .await()
+                                                            .results;
+        if (similarBusinessesInTheArea.length > 1){
+          checkBusinessThroughLinkedin(currentBusiness.getName(), similarBusinessesInTheArea);
+        }
+      }
+    } catch(GeneralSecurityException e) {
+      LOGGER.warning(e.getMessage());
+    } catch(IOException e) {
+      LOGGER.warning(e.getMessage());
+    } catch(ApiException e) {
+      LOGGER.warning(e.getMessage());
+    } catch(InterruptedException e) {
+      LOGGER.warning(e.getMessage());
```

Comment:
1. So I moved the try and catch for only the await method call. However, I am not sure what is throwing the GeneralSecurityException, I can not find it online, but it might be the documenetaion is outdated a bit, because if I remove this the program tell me I need it. 
2. Thank you for this suggestion